analytics-integrations/moengage/src/test/java/com/segment/analytics/internal/integrations/MoEngageTest.java

```diff
@@ -0,0 +1,141 @@
+public class MoEngageTest {
+
+    @Rule public PowerMockRule rule = new PowerMockRule();
+    @Rule public IntegrationTestRule integrationTestRule = new IntegrationTestRule();
+
+    @Mock Application application;
+    @Mock Context context;
+    @Mock Analytics analytics;
+    MoEngageIntegration integration;
+
+    @Before
+    public void setUp() {
+        initMocks(this);
+        integration = new MoEngageIntegration();
+        integration.mHelper = MoEHelper.getInstance(application);
+    }
+
+    @Test public void initialize(){
+        //nothing to be done
+    }
+
+    @Test public void activityCreate() {
+        Activity activity = mock(Activity.class);
+        Bundle bundle = mock(Bundle.class);
+        integration.onActivityCreated(activity, bundle);
+        verifyNoMoreInteractions(MoEHelper.class);
+    }
+
+    @Test public void activityStart() {
+        Activity activity = mock(Activity.class);
+        integration.onActivityStarted(activity);
+        verifyNoMoreInteractions(MoEHelper.class);
+    }
+
+    @Test public void activityResume() {
+        Activity activity = mock(Activity.class);
+        integration.onActivityResumed(activity);
+        verifyNoMoreInteractions(MoEHelper.class);
+    }
+
+    @Test public void activityPause() {
+        Activity activity = mock(Activity.class);
+        integration.onActivityPaused(activity);
+        verifyNoMoreInteractions(MoEHelper.class);
+    }
+
+    @Test public void activityStop() {
+        Activity activity = mock(Activity.class);
+        integration.onActivityStopped(activity);
+        verifyNoMoreInteractions(MoEHelper.class);
+    }
+
+    @Test public void activityDestroy(){
+        //do nothing
+    }
+
+    @Test public void group() {
+        integration.group(new GroupPayloadBuilder().build());
+        verifyNoMoreInteractions(MoEHelper.class);
+    }
+
+    @Test public void alias() {
+        integration.alias(new AliasPayloadBuilder().build());
+        verifyNoMoreInteractions(MoEHelper.class);
+    }
+    @Test
+    public void activitySaveInstance() {
+        Activity activity = mock(Activity.class);
+        Bundle bundle = mock(Bundle.class);
+        integration.onActivitySaveInstanceState(activity, bundle);
+        verifyNoMoreInteractions(MoEHelper.class);
+    }
+
+
+    @Test public void track() {
+        Activity activity = mock(Activity.class);
+        integration.track(new TrackPayloadBuilder().event("foo").build());
+        verifyNoMoreInteractions(MoEHelper.class);
+    }
+
+    @Test public void screen() {
+        integration.screen(new ScreenPayloadBuilder().build());
+        verifyNoMoreInteractions(MoEHelper.class);
+    }
+
+    @Test public void identify() {
+        Traits traits = TestUtils.createTraits("foo");
+        traits.putEmail("foo@bar.com");
+        integration.identify(new IdentifyPayloadBuilder().traits(traits).build());
+        verifyNoMoreInteractions(MoEHelper.class);
```

Comment:
Verify `mHelper. setUserLocation` and `mHelper.setUserAttribute` were called in the correct conditions.